File: rides-service/src/main/java/com/example/rides_service/client/PassengerServiceClient.java
package com.example.rides_service.client;

import com.example.rides_service.config.RetrieveMessageErrorDecoder;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "passengerService", url = "${feign.client.passenger-service-url}", configuration = RetrieveMessageErrorDecoder.class)
public interface PassengerServiceClient {

    @GetMapping("/{id}/exists")
    ResponseEntity<Boolean> doesPassengerExists(@PathVariable("id") Long passengerId);

}

File: rating-service/src/main/kotlin/com/example/rating_service/client/PassengerServiceClient.kt
package com.example.rating_service.client

import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker
import io.github.resilience4j.retry.annotation.Retry
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable

@FeignClient(
    name = "passengerService",
    url = "\${feign.client.passenger-service-url}",
    fallback = PassengerServiceClientFallback::class,
    configuration = [RetrieveMessageErrorDecoder::class]
)
interface PassengerServiceClient {
    @GetMapping("/{id}/exists")
    @CircuitBreaker(name = "passengerClient")
    @Retry(name = "passengerClientRetry")
    fun doesPassengerExist(@PathVariable("id") passengerId: Long): ResponseEntity<Boolean>

}